src/blockdata/block.rs

```diff
@@ -135,27 +82,21 @@ impl BlockHeader {
     }
 
     /// Checks that the proof-of-work for the block is valid, returning the block hash.
-    pub fn validate_pow(&self, required_target: &Uint256) -> Result<BlockHash, util::Error> {
+    pub fn validate_pow(&self, required_target: &Target) -> Result<BlockHash, util::Error> {
```

Comment:
Pass by value would feel much more natural here.

Reply:
Just noticed this one, done as suggested.